RepairsListDetailsView

vanyalaptii opened the pull request.

vanyalaptii committed:
RepairListDetailView

Added a view with fields for reading, editing information and saving it to database.

vanyalaptii committed:
Fixed UI list update

Was fixed problem when after repair fields changed and back button pressed repairs list not update repair info.

OksanaFedorchuk commented on ReFix/Modules/RepairsView/RepairsListViewModel.swift:
```diff
@@ -0,0 +1,101 @@
+    @Published var addNewRepairIsPresented: Bool = false
+    @Published var repairDatailIsPresented: Bool = false
+    
+    @Published var brand: String = ""
```
ти використовуєш всі ці проперті в лісті?

vanyalaptii replied:
Виправлено

vanyalaptii committed:
Merge branch 'main' into feature/RepairsListDetailsView

vanyalaptii committed:
Merge changes from main

Merge changes from main.

OksanaFedorchuk commented on ReFix/Modules/Repairs/RepairDetailView.swift:
```diff
@@ -0,0 +1,80 @@
+    }
+}
+
+//#Preview {
```
Remove or fix?

vanyalaptii replied:
виправлено

OksanaFedorchuk commented on ReFix/Modules/Repairs/RepairDetailViewModel.swift:
```diff
@@ -0,0 +1,84 @@
+        
+        self.repair = repair
+        
+        self.id = repair.wrappedValue.id
```
Is this a duplication of  func cleanUp() ?

vanyalaptii replied:
змінив назву

OksanaFedorchuk replied:
в тебе є 
RepairDetailView
        .onDisappear() {
            viewModel.discardUnsavedValues()
        } 
        
який присвоює значення repair всім проперті

якщо ти хочеш засінкати їх ще на ініті RepairDetailViewModel, ти можеш реюзнути цей же метод
Наприклад, викликати його теж на 
RepairDetailView
        .onAppear() {
            viewModel.discardUnsavedValues()
        } 
        
і подумай над неймінгом, може щось на кшталт, inectRepairValues() чи щось більш загальне, бо ти не діскардиш нічого і не витираєш, ти присвоюєш значення

OksanaFedorchuk commented on ReFix/Modules/Repairs/RepairsListView.swift:
```diff
@@ -67,15 +32,45 @@ struct RelpairsListView: View {
+        }
+        .padding()
+        .font(.largeTitle)
+        .sheet(isPresented: $viewModel.isAddNewRepairPresented) {
```
move this on main body

vanyalaptii replied:
виправив

OksanaFedorchuk replied:
а пушнув? не бачу зміни

OksanaFedorchuk commented on ReFix/Modules/Repairs/RepairDetailView.swift:
```diff
@@ -0,0 +1,80 @@
+import SwiftUI
+
+struct RepairDetailView: View {
+    enum Field: Hashable {
```
is this private?

vanyalaptii replied:
виправив

OksanaFedorchuk commented on ReFix/Modules/Repairs/RepairDetailView.swift:
```diff
@@ -0,0 +1,80 @@
+    }
+    
+    @EnvironmentObject private var viewModel: RepairDetailViewModel
+    @FocusState var isFocused: Field?
```
is this private?

vanyalaptii replied:
виправив

OksanaFedorchuk replied:
isFocused в тебе не використовується

OksanaFedorchuk commented on ReFix/Modules/Repairs/RepairDetailView.swift:
```diff
@@ -0,0 +1,80 @@
+    }
+}
+
+extension RepairDetailView {
```
are these private?

OksanaFedorchuk replied:
(also, better to add markers for extensions)

vanyalaptii replied:
виправив

OksanaFedorchuk replied:
better private extension{
here all becomes private automatically
}

OksanaFedorchuk commented on ReFix/Modules/Repairs/RepairsListView.swift:
```diff
@@ -67,15 +32,45 @@ struct RelpairsListView: View {
+            RoundedRectangle(cornerRadius: 10, style: .continuous)
+                .stroke(Color(uiColor: .tertiaryLabel), lineWidth: 1)
+        }
+        .listRowSeparator(.hidden)
```
better to use list setups in main body (line 17)

vanyalaptii replied:
it is not working

OksanaFedorchuk replied:
<img width="598" alt="Screenshot 2024-03-02 at 10 25 41" src="https://github.com/vanyalaptii/ReFix/assets/51946521/c7151f61-f4ba-4a5a-adb7-1e91c84eb9df">

OksanaFedorchuk commented on ReFix/Modules/Repairs/RepairsListViewModel.swift:
```diff
@@ -18,26 +18,31 @@ final class RepairsListViewModel: ObservableObject {
-                .sorted { $0.id > $1.id }
-//                 + repairsArray.filter { $0.client.name.contains(searchText) }
-//                 + repairsArray.filter { $0.client.phoneNumber.contains(searchText)}
+        get {
```
this holds too much logic for a computed property, move to a method

vanyalaptii replied:
виправив

OksanaFedorchuk reviewed (commented):
Go through view models and mark private methods as private, better move them to extensions

vanyalaptii committed:
Access control

1. Properties and functions marked as private.
2. cleanUp() function renamed to discardUnsavedValues().
3. Search logic moved to searchResult() method.
